src/reconnectcontroller/DefaultReconnectController.ts

```diff
@@ -26,7 +28,9 @@ export default class DefaultReconnectController implements ReconnectController {
   }
 
   private hasPastReconnectDeadline(): boolean {
```

Comment:
- `hasPastReconnectDeadline()` is being used only by `retryWithBackoff()`.
- Let's take a look at `retryWithBackoff()`. In addition to `hasPastReconnectDeadline()`, it's using `this.shouldReconnect` to determine whether or not to retry, right?

```
  retryWithBackoff(retryFunc: () => void, cancelFunc: () => void): boolean {
    const willRetry = this.shouldReconnect && !this.hasPastReconnectDeadline();
    if (willRetry) {
   ...
```

- Look at `disableReconnect()`. This is a public method.
```
  disableReconnect(): void {
    this.shouldReconnect = false;
  }
```

In summary, you can disable the reconnect operation without modifying the `ReconnectController` interface and the `DefaultReconnectController` implementation.

Note that `reconnectController` is already available in `AudioVideoControllerState` so you can disable the reconnect operation literally anywhere in the audio-video controller, in tasks, or even in your component.